bases/node-ts.json

```
@@ -0,0 +1,15 @@
+{
+    "$schema": "https://json.schemastore.org/tsconfig",
+    "display": "Node with TypeScript (TS >=5.8 ONLY)",
+    "docs": "https://devblogs.microsoft.com/typescript/announcing-typescript-5-8-beta/#the---erasablesyntaxonly-option",
+    "_version": "23.6.0",
+    "compilerOptions": {
+        "module": "node16",
+        "target": "es2024",
+        "strict": true,
+        "esModuleInterop": true,
+        "skipLibCheck": true,
+        "moduleResolution": "node16",
+        "erasableSyntaxOnly": true
```

```suggestion
        "erasableSyntaxOnly": true
```

I was actually sort of thinking this, but removing options already set by say, `@types/node22`, that leaves this config empty, which I think is pretty weird, so maybe this is a bad idea and we should just have this be more fleshed out. Not sure...

---

I originally thought this as well, but then in this case the file would always need to be extended from another file, if we keep the settings, it becomes standalone but also extendable as long as it comes first in the stack

---

Extends is an array in modern typescript right? 

So it could be made to expect `["@tsconfig/bases/node22", "@tsconfig/bases/nodeStripTypes"]`  or similar?

---

Yes that was our idea originally, but then if someone wants to use it as a standalone without any other extends it will also have it ready (or we can pre-add the extends)

---

I was thinking about the previous response of adding the extends here, but I think this wouldn't work because we would reference a local file, and I think when the NPM package is downloaded it doesn't contain all the bases so it would fail

---

Not in the tsconfig base, but as a recommendation that users write both in _their_ tsconfig

---

@orta @jakebailey what do you think?